.gitlab/docker/Dockerfile.spack_build

```diff
@@ -0,0 +1,17 @@
+ARG BASE_IMAGE
+FROM $BASE_IMAGE
+
+ARG SOURCE_DIR=/pika/source
+ARG BUILD_DIR=/pika/build
+
+ARG CMAKE_COMMON_FLAGS
+ARG CMAKE_FLAGS
+
+ENV DEBIAN_FRONTEND=noninteractive
```

Comment:
Necessary? It's already set, and you don't use apt anymore after the base image right?